integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchema.scala

```diff
@@ -774,8 +774,8 @@ case class LoadTable(
         GlobalDictionaryUtil.updateTableMetadataFunc = LoadTable.updateTableMetadata
         val storePath = relation.tableMeta.storePath
         // add the start entry for the new load in the table status file
-        CommonUtil.
-          readAndUpdateLoadProgressInTableMeta(carbonLoadModel, storePath, isOverwriteTable)
+        CommonUtil.readAndUpdateLoadProgressInTableMeta(carbonLoadModel, storePath,
```

Comment:
Here we can add if check like below
```
if (!updateModel.isDefined) {
CommonUtil.readAndUpdateLoadProgressInTableMeta(carbonLoadModel, storePath, isOverwriteTable)
}
```

Reply:
Agree. But it need to read 'LoadMetadataDetails' when updateModel.isDefined == true, otherwise it will throw NPE because carbonLoadModel.getLoadMetadataDetails is null.
Add code like below:

```
        if (!updateModel.isDefined) {
          CommonUtil.
            readAndUpdateLoadProgressInTableMeta(carbonLoadModel, storePath, isOverwriteTable)
        } else {
          CommonUtil.readLoadMetadataDetails(carbonLoadModel, storePath)
        }
```
@ravipesala , it don't need to add a load meta recorder when execute update, right?